src/main/java/seedu/taskitty/logic/ToolTip.java

```diff
@@ -0,0 +1,202 @@
+        commands.setPredicate(p -> p.startsWith(command));
+        
+        if (!isCommandWordMatch()) {
+            setToolTip(MESSAGE_UNKNOWN_COMMAND);
+        } else if (isSingleMatchFound()) {
+            getMatchCommandToolTipSingle(command);
+        } else {
+            getMatchCommandToolTipAll();
+        }
+    }
+    
+    /**
+     * Returns true if there is at least 1 command word that matches
+     */
+    private boolean isCommandWordMatch() {
+        if (commands.size() != COMMAND_WORD_COUNT_NO_MATCH) {
+            return true;
+        }
+        return false;
+    }
+    
+    /**
+     * Returns true if there is exactly 1 command word that matches
+     */
+    private boolean isSingleMatchFound() {
+        if (commands.size() == COMMAND_WORD_COUNT_SINGLE_MATCH) {
+            return true;
+        }
+        return false;
+    }
+    
+    /**
+     * Finds the closest matching command and returns the appropriate tooltip
+     *  
+     * @param command to determine which command tooltip to show
+     */
+    private void getMatchCommandToolTipSingle(String command) {
+        if (AddCommand.COMMAND_WORD.startsWith(command)) {
+            
+            setToolTip(AddCommand.MESSAGE_PARAMETER, AddCommand.MESSAGE_USAGE);
+            
+        } else if (ViewCommand.COMMAND_WORD.startsWith(command)) {
+            
+            setToolTip(ViewCommand.MESSAGE_PARAMETER, ViewCommand.MESSAGE_USAGE);
+            
+        } else if (FindCommand.COMMAND_WORD.startsWith(command)) {
+            
+            setToolTip(FindCommand.MESSAGE_PARAMETER, FindCommand.MESSAGE_USAGE);
+            
+        } else if (EditCommand.COMMAND_WORD.startsWith(command)) {
+            
+            setToolTip(EditCommand.MESSAGE_PARAMETER, EditCommand.MESSAGE_USAGE);
+            
+        } else if (DeleteCommand.COMMAND_WORD.startsWith(command)) {
+            
+            setToolTip(DeleteCommand.MESSAGE_PARAMETER, DeleteCommand.MESSAGE_USAGE);
+            
+        } else if (DoneCommand.COMMAND_WORD.startsWith(command)) {
+            
+            setToolTip(DoneCommand.MESSAGE_PARAMETER, DoneCommand.MESSAGE_USAGE);
+            
+        } else if (UndoCommand.COMMAND_WORD.startsWith(command)) {
+            
+            setToolTip(UndoCommand.MESSAGE_PARAMETER, UndoCommand.MESSAGE_USAGE);
+            
+        } else if (ClearCommand.COMMAND_WORD.startsWith(command)) {
+            
+            setToolTip(ClearCommand.MESSAGE_PARAMETER, ClearCommand.MESSAGE_USAGE);
+            
+        } else if (HelpCommand.COMMAND_WORD.startsWith(command)) {
+            
+            setToolTip(HelpCommand.MESSAGE_PARAMETER, HelpCommand.MESSAGE_USAGE);
+            
+        } else if (ExitCommand.COMMAND_WORD.startsWith(command)) {
+            
+            setToolTip(ExitCommand.MESSAGE_PARAMETER, ExitCommand.MESSAGE_USAGE);
+            
+        } else {
+            setToolTip(MESSAGE_UNKNOWN_COMMAND);
+        }
```

Comment:
Since each case returns the same method, maybe you could store each message in an array or list and iterate through them to reduce duplication of code.